From ab6cb5b74a8f7d292571c219f563ee0de57e325d Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@mailbox.org>
Date: Wed, 11 Jul 2018 10:23:17 -0500
Subject: [PATCH 1/7] fix warnings for Werror

---
 amd/amdgpu/amdgpu_object.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/amd/amdgpu/amdgpu_object.c b/amd/amdgpu/amdgpu_object.c
index 1074e74..a7f9a3f 100644
--- a/amd/amdgpu/amdgpu_object.c
+++ b/amd/amdgpu/amdgpu_object.c
@@ -55,7 +55,6 @@ static void amdgpu_ttm_bo_destroy(struct ttm_buffer_object *tbo)
 {
 	struct amdgpu_device *adev = amdgpu_ttm_adev(tbo->bdev);
 	struct amdgpu_bo *bo = ttm_to_amdgpu_bo(tbo);
-	u64 offset;
 
 	if (bo->tbo.mem.mem_type == AMDGPU_PL_DGMA_IMPORT)
 		kfree(tbo->mem.bus.addr);
-- 
2.17.1

