From e2911e64985e56d79db229c86210dafee672cbba Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@mailbox.org>
Date: Wed, 11 Jul 2018 10:54:52 -0500
Subject: [PATCH 6/7] fix uninitialized use

---
 amd/amdgpu/amdgpu_vm.c | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/amd/amdgpu/amdgpu_vm.c b/amd/amdgpu/amdgpu_vm.c
index 8b5fbbc..8443bde 100644
--- a/amd/amdgpu/amdgpu_vm.c
+++ b/amd/amdgpu/amdgpu_vm.c
@@ -1442,12 +1442,18 @@ static int amdgpu_vm_bo_split_mapping(struct amdgpu_device *adev,
 		dma_addr_t *dma_addr = NULL;
 		uint64_t max_entries;
 		uint64_t addr, last;
-		uint64_t count;
+    uint64_t count;
 
 		if (nodes) {
 			addr = nodes->start << PAGE_SHIFT;
 			max_entries = (nodes->size - pfn) *
 				(PAGE_SIZE / AMDGPU_GPU_PAGE_SIZE);
+		} else {
+			addr = 0;
+			max_entries = S64_MAX;
+		}
+
+    if (pages_addr) {
 			switch (mem->mem_type) {
 			case TTM_PL_TT:
 				max_entries = min(max_entries, 16ull * 1024ull);
@@ -1489,10 +1495,7 @@ static int amdgpu_vm_bo_split_mapping(struct amdgpu_device *adev,
 		} else if (flags & AMDGPU_PTE_VALID) {
 			addr += vram_base_offset;
 			addr += pfn << PAGE_SHIFT;
-		} else {
-			addr = 0;
-			max_entries = S64_MAX;
-		}
+    }
 
 		last = min((uint64_t)mapping->last, start + max_entries - 1);
 		r = amdgpu_vm_bo_update_mapping(adev, exclusive, dma_addr, vm,
-- 
2.17.1

