Only in lightdm-gtk-greeter-1.8.3-b: config.log
diff -ur lightdm-gtk-greeter-1.8.3-a/src/lightdm-gtk-greeter.c lightdm-gtk-greeter-1.8.3-b/src/lightdm-gtk-greeter.c
--- lightdm-gtk-greeter-1.8.3-a/src/lightdm-gtk-greeter.c	2014-03-13 05:17:37.000000000 -0500
+++ lightdm-gtk-greeter-1.8.3-b/src/lightdm-gtk-greeter.c	2014-03-26 08:17:16.587227462 -0500
@@ -2305,10 +2305,19 @@
     ido_init ();
 #endif
 
+    char* config_file = getenv("LIGHTDM_GTK_GREETER_CONF");
+
     config = g_key_file_new ();
-    g_key_file_load_from_file (config, CONFIG_FILE, G_KEY_FILE_NONE, &error);
-    if (error && !g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
-        g_warning ("Failed to load configuration from %s: %s\n", CONFIG_FILE, error->message);
+    if (config_file != 0)
+    {
+      g_key_file_load_from_file (config, config_file, G_KEY_FILE_NONE, &error);
+      if (error && !g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+          g_warning ("Failed to load configuration from %s: %s\n", config_file, error->message);
+    } else {
+      g_key_file_load_from_file (config, CONFIG_FILE, G_KEY_FILE_NONE, &error);
+      if (error && !g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+          g_warning ("Failed to load configuration from %s: %s\n", CONFIG_FILE, error->message);
+    }
     g_clear_error (&error);
 
     state_dir = g_build_filename (g_get_user_cache_dir (), "lightdm-gtk-greeter", NULL);
