From 2b25f18f1ced8b716d6d1e94ec3cd18849543897 Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@mailbox.org>
Date: Fri, 20 Oct 2017 06:19:02 -0500
Subject: [PATCH 09/11] startplasma: Don't use xmessage

---
 startkde/startplasma.cmake | 2 --
 1 file changed, 2 deletions(-)

diff --git a/startkde/startplasma.cmake b/startkde/startplasma.cmake
index 6f771d8e..e1c8fff6 100644
--- a/startkde/startplasma.cmake
+++ b/startkde/startplasma.cmake
@@ -71,7 +71,6 @@ if test $? -ne 0; then
   # Startup error
   echo 'startplasma: Could not start kdeinit5. Check your installation.'  1>&2
   test -n "$ksplash_pid" && kill "$ksplash_pid" 2>/dev/null
-  @NIXPKGS_XMESSAGE@ -geometry 500x100 "Could not start kdeinit5. Check your installation."
   exit 1
 fi
 
@@ -95,7 +94,6 @@ if test $? -eq 255; then
   # Startup error
   echo 'startplasma: Could not start ksmserver. Check your installation.'  1>&2
   test -n "$ksplash_pid" && kill "$ksplash_pid" 2>/dev/null
-  @NIXPKGS_XMESSAGE@ -geometry 500x100 "Could not start ksmserver. Check your installation."
 fi
 
 wait_drkonqi=$(@NIXPKGS_KREADCONFIG5@ --file startkderc --group WaitForDrKonqi --key Enabled --default true)
-- 
2.16.1

